selectors/selectors.md

```diff
@@ -108,3 +108,269 @@ These components imply or expand into the following things:
 - Allow language-specific implementations of selectors (parsers, execution, etc).
 - (IMPORTANT) Well-designed set of test vectors representing a variety of use cases for IPLD Selectors.
 - A recommended structure for implementing a selector type, with an easy to use test suite.
+
+
+Selector types
+--------------
+
+### CID Selector
+
+Selects a single CID.
+
+
+### Path Selector
+
+Selects a single path segment.
+
+
+### Array Selector
+
+Selects a slice or single item of an array of items.
```

Comment:
Actually, isn't this more of a "range" selector? That is, we're selecting a "range" of keys (from an object or an array).

Reply:
That's actually a good point. I named it after the data structures it operates on. But it makes sense to name them after the result (what they actually return). Is it OK if I create a PR once this one is merged (so that we can then discuss it there if anyone has objections, instead of postponing this one even more).

Reply:
Actually this part is outdated. We don't have a general "Array Selector" anymore. We have a single item selector, slice and all elements selector now. I'll update the spec.

Reply:
https://github.com/ipld/specs/pull/104